### DISCUSSION TITLE
Download a File documentWithCaptionMessage

### DISCUSSION THREAD

**Discussion by pierreoramassyntegra** [WhiskeySockets/Baileys#342] [General]
I try to Download a PDF File sended with Caption.

Then the message type is set as **documentWithCaptionMessage**.

When send me a PDF without Caption the message is set to **documentMessage** and work fine.

But when the type is set to **documentWithCaptionMessage**, not work.

This is the JSon Message when the type is set to **documentWithCaptionMessage**:

{
	"key": {
		"remoteJid": "593963488439@s.whatsapp.net",
		"fromMe": false,
		"id": "3EB0B2C0229BD7F28FEBAE"
	},
	"messageTimestamp": 1692647216,
	"pushName": "Tele",
	"broadcast": false,
	"message": {
		"messageContextInfo": {
			"deviceListMetadata": {
				"senderKeyHash": "5Xj/hSlBA9o9wA==",
				"senderTimestamp": "1692625481",
				"recipientKeyHash": "aG5Fu7DNsqja5A==",
				"recipientTimestamp": "1692306519"
			},
			"deviceListMetadataVersion": 2
		},
		"documentWithCaptionMessage": {
			"message": {
				"documentMessage": {
					"url": "https://mmg.whatsapp.net/v/t62.7119-24/29390064_844205120658640_3485801630009598824_n.enc?ccb=11-4&oh=01_AdQ6hkqfPQJQdqLrUbedmdm2VUkyIJEzgEh5g2S9pWbMNg&oe=650B22FF&mms3=true",
					"mimetype": "application/pdf",
					"title": "How to set up TypeScript with Node.js and Express.pdf",
					"fileSha256": "YDZ1Gs5pQUcRfcDppsexE8ddLjMj1vDGnpjoWLi0wt8=",
					"fileLength": "627219",
					"pageCount": 16,
					"mediaKey": "8IhjgM5obZLsknbK2nPAv0E3E5VMT9LMdpwTFrNcMww=",
					"fileName": "How to set up TypeScript with Node.js and Express.pdf",
					"fileEncSha256": "88V6skp++ZKtQm/wvPij/XOnMynd/K8/y2vv3KErvcs=",
					"directPath": "/v/t62.7119-24/29390064_844205120658640_3485801630009598824_n.enc?ccb=11-4&oh=01_AdQ6hkqfPQJQdqLrUbedmdm2VUkyIJEzgEh5g2S9pWbMNg&oe=650B22FF",
					"mediaKeyTimestamp": "1692646478",
					"contactVcard": false,
					"thumbnailDirectPath": "/v/t62.36145-24/30083608_162572813533867_370795425461295503_n.enc?ccb=11-4&oh=01_AdRnTOOGkrskix50PM8i1cbegbubUmFlsFH15UmXkXgegw&oe=650B2154",
					"thumbnailSha256": "i79bnce9JRqib8fRgvijfH2nXhvCHy/mtyE+oN6Xn3U=",
					"thumbnailEncSha256": "kQaDzDI9S8PYqz+mT3loQ+CWvm35rsLXbd1kEyF/77Y=",
					"jpegThumbnail": "/9j/4AAQSkZJRgABAQAAAQABAAD/4gHYSUNDX1BST0ZJTEUAAQEAAAHIAAAAAAQwAABtbnRyUkdCIFhZWiAH4AABAAEAAAAAAABhY3NwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAA9tYAAQAAAADTLQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAlkZXNjAAAA8AAAACRyWFlaAAABFAAAABRnWFlaAAABKAAAABRiWFlaAAABPAAAABR3dHB0AAABUAAAABRyVFJDAAABZAAAAChnVFJDAAABZAAAAChiVFJDAAABZAAAAChjcHJ0AAABjAAAADxtbHVjAAAAAAAAAAEAAAAMZW5VUwAAAAgAAAAcAHMAUgBHAEJYWVogAAAAAAAAb6IAADj1AAADkFhZWiAAAAAAAABimQAAt4UAABjaWFlaIAAAAAAAACSgAAAPhAAAts9YWVogAAAAAAAA9tYAAQAAAADTLXBhcmEAAAAAAAQAAAACZmYAAPKnAAANWQAAE9AAAApbAAAAAAAAAABtbHVjAAAAAAAAAAEAAAAMZW5VUwAAACAAAAAcAEcAbwBvAGcAbABlACAASQBuAGMALgAgADIAMAAxADb/2wBDAAMCAgICAgMCAgIDAwMDBAYEBAQEBAgGBgUGCQgKCgkICQkKDA8MCgsOCwkJDRENDg8QEBEQCgwSExIQEw8QEBD/2wBDAQMDAwQDBAgEBAgQCwkLEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBD/wAARCAAaABIDASIAAhEBAxEB/8QAGQABAAMBAQAAAAAAAAAAAAAAAAUGCAMJ/8QAJRAAAgICAgEEAgMAAAAAAAAAAQIDBAURAAYSEyExMgcUFUFi/8QAFgEBAQEAAAAAAAAAAAAAAAAAAQAD/8QAGBEBAQEBAQAAAAAAAAAAAAAAAAECERL/2gAMAwEAAhEDEQA/APSSjcyNXKZnrw61mHEEhs423kI6iY7ITymSxqN6oeSERORGz2IkYkKy+s3k7WOxjKMrVZv4unLLSsmzXMiAGCVw6SSofElXKSyjY0SHYEgMeVLsHfOo4bNZImlWvXevQpPeaG9jY5KrzCNVEnrzo8TtGQwL+Ksin3J8AZbD996/mr0OLr2q6XJo2P65yFSSRZo3dLEBWKViZIHjKyaBVSdBiQwVt7epY+OOOCQlrG4rLX7PrS5pJlC05PStXa0WtLMCngyofgAyJ/qMt9l5I4/HV8ZCYK0lp1ZvImxalsNvQH2kZiBoD23rez8kk4+/P9mzT/JtxKliSBbk2ceyI3KiZo8PRSMvr7FUllVSfhZHA0GO9j169epXjq1II4YIUEcccahURQNBQB7AAf0Oabx5kpsdOOOOZh//2Q==",
					"contextInfo": {
						"disappearingMode": {
							"initiator": "CHANGED_IN_CHAT"
						}
					},
					"thumbnailHeight": 480,
					"thumbnailWidth": 339,
					"caption": "Cuarto Documento"
				}
			}
		}
	},
	"verifiedBizName": "Tele"
}

**Comment by Rippanteq7** [comment]
![IMG_20230822_122010](https://github.com/WhiskeySockets/Baileys/assets/104727253/5d2e50c6-2e40-43cc-ac84-bd69a89f75b5)

Use ```downloadMediaMessage``` function, it's working fine for me

**Reply by pierreoramassyntegra** [reply]
I currently create a new object WAMessage, converting the documnetWithCaptionMessage in a documentMessage.

var msg2:WAMessage={
		key:msg.key,
		messageTimestamp:msg.messageTimestamp,
		pushName:msg.pushName,
		broadcast:msg.broadcast,
		message:{
			documentMessage:msg.message.documentWithCaptionMessage.message.documentMessage,
			messageContextInfo:msg.message.messageContextInfo
		},
		verifiedBizName:msg.verifiedBizName
	};

Then use this documentMessage for be downladed the same way as a traditional documentMessage.

const buffer = await downloadMediaMessage(
		msg2,
		'buffer',
		{ },
		{ 
			logger,
			// pass this so that baileys can request a reupload of media
			// that has been deleted
			reuploadRequest: sock.updateMediaMessage
		}
	)

**Comment by pierreoramassyntegra** [comment]
After I found a way in my previous message. Then I will to close this discussion
